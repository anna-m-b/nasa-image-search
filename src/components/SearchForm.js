import React from 'react';
import PropTypes from 'prop-types';
import '../styles/SearchForm.css';

const SearchForm = ({ onChange, onSearchSubmit, searchInput }) => {
  return (
    <form className="search-form" onSubmit={onSearchSubmit}>
      <label htmlFor="search-input" className="search-form_label">
        Search the NASA images database
      </label>
      <span className="search-form_input-and-button">
        <input
          type="text"
          placeholder="Enter a search term"
          id="search-input"
          className="search-form_input"
          value={searchInput}
          onChange={onChange}
        />

        <button type="submit" className="search-form_button">
          Search
        </button>
      </span>
    </form>
  );
};

SearchForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSearchSubmit: PropTypes.func.isRequired,
  searchInput: PropTypes.string.isRequired
}


export default SearchForm;
