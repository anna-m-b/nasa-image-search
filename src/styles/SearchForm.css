@import url('https://fonts.googleapis.com/css2?family=Oxygen:wght@400;700&display=swap');

.search-form {
  width: 60%;
  margin: 0 auto;
  display: grid;
  grid-template: 1fr 1fr / 1fr;
  justify-content: center;
}

.search-form_label {
 margin: 15px;
 font-style: italic;
 color: grey;
}

.search-form_input {
  width: 40%;
  height: 28px;
  border-radius: 5px;
  padding: 0 2px 0 5px;
  vertical-align: middle;
}

.search-form_button {
  width: 100px;
  height: 30px;
  padding: 0 10px 0 10px;
  margin-left: 3px;
  border: none;
  border-radius: 5px;
  background-color: #e60000;
  color: white;
  text-transform: uppercase;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: 2px;
  vertical-align: middle;
}