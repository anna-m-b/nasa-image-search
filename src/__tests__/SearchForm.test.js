import React from 'react';
import { render } from '@testing-library/react';
import SearchForm from '../components/SearchForm';

describe('Search', () => {
  const { asFragment } = render(<SearchForm />);

  test('renders correctly', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
