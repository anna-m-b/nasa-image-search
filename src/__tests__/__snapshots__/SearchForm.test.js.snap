// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <label
      class="search-form_label"
      for="search-input"
    >
      Search the NASA images database
    </label>
    <span
      class="search-form_input-and-button"
    >
      <input
        class="search-form_input"
        id="search-input"
        placeholder="Enter a search term"
        type="text"
        value=""
      />
      <button
        class="search-form_button"
        type="submit"
      >
        Search
      </button>
    </span>
  </form>
</DocumentFragment>
`;
